src/Main.elm

```diff
@@ -1,10 +1,251 @@
+                        style "opacity" "0.5"
+
+                      else
+                        style "opacity" "1"
+                    ]
+                    [ text (Debug.toString type_) ]
+            )
+            allTypes
+            |> div [ class "button-group" ]
+
+        --folders are handled here
+        , List.map
+            (\folder ->
+                let
+                    --is the current folder opened? (only one folder can be opened at a time)
+                    opened =
+                        case model.currentFolderId of
+                            Just id ->
+                                if folder.id == id then
+                                    True
+
+                                else
+                                    False
+
+                            Nothing ->
+                                False
+                in
+                div
+                    []
+                    [ Data.AudienceFolder.view folder opened (Select folder.id) (GoUp folder.parent)
+                    ]
+            )
+            (case model.filter of
+                Data.Audience.Shared ->
+                    []
+
+                _ ->
+                    filterAudienceFolders model.folders model.currentFolderId
+            )
+            |> div []
+
+        --audiences are handled here
+        , div []
+            [ List.map
+                (\audience ->
+                    Data.Audience.view audience
+                )
+                (case model.filter of
+                    Data.Audience.Shared ->
+                        filterAudienceCategory model.audiences Data.Audience.Shared
+
+                    filterValue ->
+                        filterAudiences (filterAudienceCategory model.audiences filterValue) model.currentFolderId
+                )
+                |> div []
+            ]
+        ]
+
+
+filterAudienceCategory : List Audience -> Data.Audience.AudienceType -> List Audience
+filterAudienceCategory audiences type_ =
+    {- | Helper function to use with List.filter, separate audiences by type_ -}
+    List.filter
+        (\e ->
+            if e.type_ == type_ then
+                True
+
+            else
+                False
+        )
+        audiences
+
+
+filterAudienceFolders : List AudienceFolder -> Maybe Int -> List AudienceFolder
+filterAudienceFolders folders currentFolderId =
+    {- | Filter all eligible folders that should be displayed for the current level. -}
+    List.filter
+        (\e ->
+            case currentFolderId of
+                Just val ->
+                    if e.id == val then
+                        True
+
+                    else
+                        case e.parent of
+                            Just parentID ->
+                                if parentID == val then
+                                    True
+
+                                else
+                                    False
```

Comment:
Again, redundant if .. else, you are already returning a boolean!
```suggestion
                                parentID == val
```

Reply:
Even more simplification, only `Just val == e.parent` is enough